language/go/resolve.go

```diff
@@ -288,11 +290,20 @@ func resolveExternal(moduleMode bool, rc *repo.RemoteCache, imp string) (label.L
 		pkg = pathtools.TrimPrefix(impWithoutSemver, prefix)
 	}
 
-	return label.New(repo, pkg, defaultLibName), nil
+	nc := gc.goNamingConvention
+	if gr, ok := c.GoRepos[repo]; ok {
```

Comment:
Instead of `GoRepos`, find the matching rule in `c.Repos`, then `r.AttrString("build_naming_convention")`.

Optionally, create a `map[string]namingConvention` to speed up the lookup, and store it in `goConfig`. `Configure` is probably the right place to do that, when `rel == ""` (in the root directory).